src/Containers/ProductDetail.js

```diff
@@ -0,0 +1,78 @@
+import React, { useEffect, useCallback } from 'react';
+import { useParams } from 'react-router-dom';
+import axios from 'axios';
+import { useDispatch, useSelector } from 'react-redux';
+import { selectedProduct, removeSelectedProduct } from '../Redux/Actions/productActions';
+import '../Styles/ProductDetail.css';
+
+const ProductDetail = () => {
+  const product = useSelector((state) => state.product);
+  const {
+    image, title, price, category, description, rating,
+  } = product;
+  const { productid } = useParams();
+  const dispatch = useDispatch();
+
+  const fetchProductDetail = useCallback(async () => {
+    const response = await axios.get(`https://fakestoreapi.com/products/${productid}`).catch((err) => {
+      throw new Error('Err', err);
+    });
+    dispatch(selectedProduct(response.data));
+  }, [productid, dispatch]);
+
+  useEffect(() => {
+    if (productid && productid !== '') {
+      fetchProductDetail();
+    }
+
+    return () => {
+      dispatch(removeSelectedProduct());
+    };
+  }, [productid, fetchProductDetail, dispatch]); // Include dispatch in the dependency array
+
+  return (
+    <div className="DetailsPage">
+      {Object.keys(product).length === 0 ? (
+        <div><h3>...loading</h3></div>
+      ) : (
+        <>
+          <div className="Home-header">
+            <div className="image">
+              <img className="image" src={image} alt={title} />
+            </div>
+            <h2>
+              {title}
+            </h2>
+          </div>
+          <div className="productDetail">
+            <div className="productDescription">
+              <div className="productDescription">
+                Category:
+                {category}
+              </div>
+              <div className="productDescription">
+                Product Description:
+                {description}
+              </div>
+              <div className="productDescription">
+                Price:
+                $
+                {price}
+              </div>
+              <div className="productDescription">
+                Count:
+                {rating.count}
+              </div>
+              <div className="productDescription">
+                Rating:
+                {rating.rate}
+              </div>
+            </div>
+          </div>
+        </>
+      )}
```

Comment:
Let's also ensure that the details page matches the following, as indicated by the previous design. Let's ensure that the layout/positioning of elements and color pattern closely matches the template design.

**Your design:**        |  **The origin design:**
:-------------------------:|:-------------------------:
<img width="800" alt="Your desgin" src="https://github.com/Dorcas126/capstone-project-metrics-webapp/assets/12745474/f6c3b597-f54b-4777-949b-70a84a081e34"> |  <img width="800" alt="Home Page" src="https://github.com/Dorcas126/capstone-project-metrics-webapp/assets/12745474/531dd146-f5a9-4727-95f3-efa7e745d879">